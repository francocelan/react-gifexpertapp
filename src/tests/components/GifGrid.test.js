import React from "react";
import {shallow} from "enzyme";
import { GifGrid } from "../../components/GifGrid.js";
import { useFetchGifs } from "../../hooks/useFetchGifs.js";
jest.mock('../../hooks/useFetchGifs.js');

describe("Pruebas en <GifGrid />", () => {

    const category = "Argentina";

    test("Hacemos snapshot del componente GifGrid",() => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading:true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
       
    })

    test("debe de mostrar items cuando se cargan imagenes useFetchGifs",() => {

        const gifs = [{
            id:"ABC",
            urlGif:"https://localhost/cualquier/cosa.jpg",
            title:"Cualquier cosa"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid category={category} />);
       
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);

    })


})