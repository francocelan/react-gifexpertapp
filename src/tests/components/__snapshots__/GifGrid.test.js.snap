// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Hacemos snapshot del componente GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Argentina
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Argentina
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      urlGif="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
